stdlib/socket/from_test.go

```diff
@@ -212,7 +213,7 @@ source
 			id := executetest.RandomDatasetID()
 			d := executetest.NewDataset(id)
 			c := execute.NewTableBuilderCache(executetest.UnlimitedAllocator)
-			c.SetTriggerSpec(flux.DefaultTrigger)
+			c.SetTriggerSpec(plan.DefaultTriggerSpec)
```

Comment:
I would think that sources don't need a trigger spec, since they always produces tables that are ready to be consumed downstream, right?

I question the need to keep the mapping from `kinds -> trigger spec` in a global map.  I recognize that some global maps are needed to make the language extensible, but in general I think we should avoid them where we can.

In this case, it seems like we could just make types that implement `PhysicalProcedureSpec` able to optionally implement a `TriggerSpecker` interface:
```go
type TriggerSpecker interface {
  TriggerSpec() plan.TriggerSpec
}
```
- transformations that are narrow don't implement `TriggerSpecker` and get the narrow trigger spec by default
- transformations that are not narrow can implement `TriggerSpecker` and choose whatever trigger spec they need

This would result in less boilerplate (no changes needed for narrow transformations).